### ISSUE TITLE
Webviews displaying results of an API call with Restclient extension are not accessible to screen readers when there is a lot of data

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#141529]
/CC @isidorn 

Issue Type: <b>Bug</b>

I use the Restclient extension to make API calls, as many other alternatives such as Postman are inaccessible.

In the past when using it with NVDA, the webview that opens displaying the response is accessible and can be easily read in browse mode.

Today when I tried making an API request, NVDA reads the following, and I'm completely unable to navigate the contents of the webview that contains the response:

```
vscode-webview://31320327-84c0-4ee0-bee2-44b9f42d5edc/index.html?id=31320327-84c0-4ee0-bee2-44b9f42d5edc&swVersion=4&extensionId=humao.rest-client&platform=electron&vscode-resource-base-authority=vscode-resource.vscode-webview.net&parentOrigin=vscode-file%3A%2F%2Fvscode-app
```

Given that the extension hasn't been updated for a few months and I have also not installed a new build of NVDA recently, I'm suspecting this to be a recent Vs Code regression.

VS Code version: Code - Insiders 1.64.0-insider (f93c69659de1a1466bd15e0f1c9f790e73be9208, 2022-01-25T08:42:41.055Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No
Remote OS version: Linux x64 5.10.60.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 PRO 4750U with Radeon Graphics (16 x 1697)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.23GB (16.96GB free)|
|Process Argv|--crash-reporter-id 1304d64e-abdb-456c-beb6-72c33120f731|
|Screen Reader|yes|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 5.10.60.1-microsoft-standard-WSL2|
|CPUs|AMD Ryzen 7 PRO 4750U with Radeon Graphics (16 x 1696)|
|Memory (System)|24.45GB (20.89GB free)|
|VM|0%|
</details><details><summary>Extensions (22)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-containers|ms-|0.215.0
remote-ssh|ms-|0.70.0
remote-ssh-edit|ms-|0.70.0
remote-wsl|ms-|0.64.2
vscode-remote-extensionpack|ms-|0.21.0
vscode-eslint|dba|2.2.2
dendron|den|0.79.0
dendron-markdown-links|den|0.6.20
dendron-markdown-preview-enhanced|den|0.10.57
dendron-markdown-shortcuts|den|0.12.1
dendron-paste-image|den|1.1.0
dendron-snippet-maker|den|0.1.6
gitlens|eam|11.7.0
prettier-vscode|esb|9.1.0
copilot|Git|1.7.4688
rest-client|hum|0.24.6
python|ms-|2021.12.1559732655
vscode-pylance|ms-|2022.1.3
jupyter|ms-|2022.1.1001614873
vsliveshare|ms-|1.0.5273
vsliveshare-audio|ms-|0.1.91
vsliveshare-pack|ms-|0.4.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstartedc:30421358
vscop804cf:30404767
vs360cf:30404996
vsclayoutctrc:30405799
azactmsalcf:30409288
pythonvsnew555cf:30426299
vsbas813:30426126

```

</details>

**Comment by isidorn** [comment]
@Neurrone thanks for filling this issue. Can you also reproduce with VS Code stable?
@mjbvz have there been some changes to the webview in the past couple of months that might affect its accessibility?

fyi @jooyoungseo  since I know you also use webviews

**Comment by Neurrone** [comment]
@isidorn I'll try reproducing on stable tomorrow. What's the best way of ensuring that my config in insiders is replicated to stable? Does using settings sync work across channels? 

I'd want to have stable use the same config to provide better results.

**Comment by isidorn** [comment]
@Neurrone you can enable settings sync on stable, and it should ask you what "setting sync server" to use. You should choose the insider one and that should then make sure that you have the same setting in both insiders and stable. Let me know how it goes. Thanks

**Comment by mjbvz** [comment]
We have not changed anything this iteration as far as I can remember

The most recent change I remember here is 3ccfaaff7c68ad832472a3bce609732849c54063 which sets the `role` of webview documents

**Comment by jooyoungseo** [comment]
As  both NVDA and JAWS user, I haven't experienced any noticeable change in the WebView over the last couple of months. However, it feels like the terminal a11y has been broken again.... The tabindex = "-1" has been removed from the div container where "document" role is assigned. This might cause some unanticipated issues when switching between terminal and Web View widgets.

**Comment by Neurrone** [comment]
Reproduceable in 1.63.2 stable.

I believe the contributing factor is how much data is contained in the response. I was able to trigger this when the response was 9000 lines long, while it seemed to work fine for responses that are say 1000 lines long.

Perhaps this issue has always existed but I'm noticing it now because of happening to have a response from an API that contains a lot of data.

**Comment by isidorn** [comment]
@jooyoungseo can you please file a separate issue for that
@Neurrone yes it sounds like this issue is not new. Let's assign to backlog for now and @mjbvz can investigate more when he finds the time.

**Comment by jvesouza** [comment]
I also came across the same problem.
As a workaround I set the extension to open the result in a text file.

"rest-client.previewResponseInUntitledDocument": true,

**Comment by mjbvz** [comment]
Since we haven't seen any other reports of this over the last year, I'm closing this as out-of-scope for VS Code

If someone wants to continue investigating, this sounds like it is either an issue with the rest client extension or a Chromium issue since it only happens for large documents

**Comment by Neurrone** [comment]
I can try this again and will report back on whether this is still an issue.
